gcapi/gcapi.py

```diff
@@ -197,10 +211,95 @@ class AlgorithmJobsAPI(APIBase):
     base_path = "algorithms/jobs/"
 
 
+class ChunkedUploadsAPI(APIBase):
+    base_path = "chunked-uploads/"
+
+    def _upload_file_with_exponential_backoff(self, file_info):
+        """
+        Uploads a chunk with an exponential backoff retry strategy. The maximum number of attempt is 3.
+
+        Parameters
+        ----------
+        file_info: dict
+        Contains information about the chunk, start_byte, end_byte and upload_id.
+
+        Raises
+        ------
+        ConnectionError
+            Raised if the chunk cannot be uploaded within 3 attempts.
+        """
+        filename = "whatever.bin"
```

Comment:
Last thing - the filename should come from the filename that the user has chosen to upload